queue.js

```diff
@@ -34,6 +40,18 @@ caps.on('connection', (socket) => {
     console.log('EVENT', { event: 'delivered',
       time: new Date().toString(),
       payload} );
+    let id = uuid();
+    queue.messages[id] = payload;
     caps.to(payload.store).emit('delivered', payload);
   });
+
+  socket.on('getall', () => {
+    Object.keys(queue.messages).forEach(id => {
+      socket.emit('message', { id, payload: queue.messages[id] });
+    });
+  });
+
+  socket.on('received', message => {
+    delete queue.messsages[message.id];
```

Comment:
`message.id` is causing this error: `TypeError: Cannot convert undefined or null to object`